service/src/main/java/com/codeforcommunity/processor/ImportProcessorImpl.java

```diff
@@ -61,4 +70,48 @@ public void importNeighborhoods(
       neighborhood.store();
     }
   }
+
+  @Override
+  public void importReservations(
+      JWTData userData, ImportReservationsRequest importReservationsRequest) {
+    if (userData.getPrivilegeLevel() != PrivilegeLevel.SUPER_ADMIN) {
+      throw new AuthException("User does not have the required privilege level.");
+    }
+
+    // Get superuser ID
+    Integer superAdminId = userData.getUserId();
+
+    for (ReservationImport reservationImport : importReservationsRequest.getReservations()) {
```

Comment:
So right now if there is some invalid entry halfway through the list (because a block id doesn't exist or something) then the first half of the list will be in the DB and the second half will be completely ignored. This would be a nightmare to try to rectify since then you would have to manually cut out all the reservations that were successful and remake the request body.
Instead what we can do is first translate each ReservationImport into a ReservationsRecord while checking their validity and then separately storing them once we know they're all valid. This way it's all or nothing and it's much easier to fix and rerun. This would look something like this:
```java
List<ReservationsRecord> records = importReservationsRequest.getReservations().map(reservationImport -> {
  if (...) { throw new ResourceDoesNotExistException(...) }
  ...
  ReservationsRecord reservation = db.newRecord(RESERVATIONS);
  ...
  return reservation;
});
records.forEach(record -> {
  record.store();
});
```

Reply:
Fixed